### PR TITLE
[native] Translate array agg config to velox

### PR DESCRIPTION
**Pull request by pranjalssh**
We need to add logic to presto native to translate deprecated.legacy-array-agg=true Presto config to presto.array_agg.ignore_nulls = true Velox config.

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

File: presto-native-execution/presto_cpp/main/common/Configs.h [hunk 1]
```diff
@@ -339,8 +341,14 @@ class SystemConfig : public ConfigBase {
   static constexpr std::string_view kInternalCommunicationJwtExpirationSeconds{
       "internal-communication.jwt.expiration-seconds"};
 
+  /// Use legacy version of array_agg which ignores nulls.
```
Comment (xiaoxmeng) [inline presto-native-execution/presto_cpp/main/common/Configs.h:344]:
```
nit: s/Use/Uses/
```

File: presto-native-execution/presto_cpp/main/common/Configs.h [hunk 2]
```diff
@@ -526,7 +538,15 @@ class BaseVeloxQueryConfig : public ConfigBase {
...
   BaseVeloxQueryConfig();
 
+  virtual ~BaseVeloxQueryConfig() = default;
+
+  void initialize(const std::string& filePath) override;
+
   static BaseVeloxQueryConfig* instance();
+
+ private:
+  /// Update velox config with values from presto system config.
```
Comment (xiaoxmeng) [inline presto-native-execution/presto_cpp/main/common/Configs.h:548]:
```
// for non-public comment in header.
```

File: presto-native-execution/presto_cpp/main/common/Configs.cpp [hunk 3]
```diff
@@ -660,4 +665,14 @@ BaseVeloxQueryConfig* BaseVeloxQueryConfig::instance() {
   return instance.get();
 }
 
+void BaseVeloxQueryConfig::initialize(const std::string& filePath) {
+  ConfigBase::initialize(filePath);
+  update(*SystemConfig::instance());
```
Comment (xiaoxmeng) [inline presto-native-execution/presto_cpp/main/common/Configs.cpp:670]: Do we need to update here if have called update(*SystemConfig::instance()) in ctor? Thanks!

Also can we test this in BaseVeloxQueryConfigTest? Thanks!

**Review by xiaoxmeng — APPROVED** [review]
@pranjalssh thanks for the change % comments. Could we add a test to capture this behavior? Thanks!

File: presto-native-execution/presto_cpp/main/common/Configs.cpp (hunk 3 above)
Comment (pranjalssh) [reply presto-native-execution/presto_cpp/main/common/Configs.cpp:670]: Initial ctor will load from default values. SystemConfig may be initialized from a file after BaseVeloxQueryConfig is created - so we load again.

**Pull request merged**